src/applications/disability-benefits/2346/components/AddressViewField.jsx

```diff
@@ -42,25 +67,55 @@ const AddressViewField = ({ formData }) => {
 
   return (
     <>
-      {street && city && country ? (
-        <div className="vads-u-border-left--7px vads-u-border-color--primary">
-          <p className="vads-u-margin-left--2 vads-u-margin-top--0">
-            {addLine(street)}
-            {addLine(street2)}
-            {addLine(street3)}
-            {country === 'USA'
-              ? `${city}, ${state} ${postalString}`
-              : `${city}, ${province} ${internationalPostalCode}`}
-            <br />
-            {country}
+      {addressType === 'permanentAddress' &&
+        !isAddressMissing && (
+          <div className="vads-u-border-left--7px vads-u-border-color--primary">
+            <p className="vads-u-margin-left--2 vads-u-margin-top--0">
+              {isBaseAddressDataValid && (
+                <>
+                  {addLine(street)}
+                  {addLine(street2)}
+                  {addLine(street3)}
+                  {isDomesticAddressValid &&
+                    `${city}, ${state} ${postalString}`}
+                  {isInternationalAddressValid &&
+                    `${city}, ${province} ${internationalPostalCode}`}
+                  <br />
+                  {country}
+                </>
+              )}
+            </p>
+          </div>
+        )}
+
+      {addressType === 'temporaryAddress' &&
+        !isAddressMissing && (
+          <div className="vads-u-border-left--7px vads-u-border-color--primary">
+            <p className="vads-u-margin-left--2 vads-u-margin-top--0">
+              {isBaseAddressDataValid && (
+                <>
+                  {addLine(street)}
+                  {addLine(street2)}
+                  {addLine(street3)}
+                  {isDomesticAddressValid &&
+                    `${city}, ${state} ${postalString}`}
+                  {isInternationalAddressValid &&
+                    `${city}, ${province} ${internationalPostalCode}`}
+                  <br />
```

Comment:
And this one

Reply:
Also, if these code blocks are identical with the exception of the addressType comparison, they're not needed